### DISCUSSION TITLE
Release 2.0.0b5, beta quality, for testing only

### DISCUSSION THREAD

**Discussion by ThomasWaldmann** [borgbackup/borg#7392] [General]
borgbackup 2.0.0b5 beta release - do not use this on production backup repositories.

Long changelog:
https://borgbackup.readthedocs.io/en/2.0.0b5/changes.html#version-2-0-0b5-2023-02-27

If you use pip to install this, use: pip install "borgbackup==2.0.0b5"

Docs: https://borgbackup.readthedocs.io/en/2.0.0b5/

Feedback: ...

<hr /><em>This discussion was created from the release <a href='https://github.com/borgbackup/borg/releases/tag/2.0.0b5'>Release 2.0.0b5, beta quality, for testing only</a>.</em>

**Comment by ThomasWaldmann** [comment]
If you test 2.0.0b5, see also #7283.

**Comment by ThomasWaldmann** [comment]
So, any feedback from b5 testing? Try to transfer old borg repos and all the changed stuff from changelog.

**Reply by dietmargoldbeck** [reply]
Hi Thomas,

2.0.0b5 is working much slower than 1.2 for me. A backup of my fileserver now takes 2 hours and only 1 hour  with 1.2 or 2.0.0b4. The problem is that i am backing up a few kvm images and the biggest one is most of the time active and gets retried 10 times.

The new feature of retrying active files needs an option to limit retries to a small number (or maybe including 0 for turning it off). I think an option for a maximum file size for retries would make sense. Retrying a backup of 50 MB sqlite DB is
much more likely to succeeed then 50GB KVM. But OTOH documentation and usage would be a bit difficult to explain. 

Just be able to switch the feature off would be ok for me. 

I am running 2.0.0b5  with a few test restores and no problems so far. I did not test transferring repos.
I noticed that with borg 2 the cache directories do **not** contain the name borg which is a bit annoying and i would prefer the old behaviour. 

Borg 1.2 /home/dietmar/.cache/borg/0a2cb8f22adaa78a1863b028f3d1629393ef0fdd99b3b61c11d871de9eaaee8b/README

Borg 2.0.0b5 /home/dietmar/.cache/fa673b73240ed3b75ba947fd98787fdf8ecd17204323e4a10c2754cdcc5fd547/README

**Comment by ThomasWaldmann** [comment]
> 2.0.0b5 is working much slower than 1.2 for me. A backup of my 
> fileserver now takes 2 hours and only 1 hour with 1.2 or 2.0.0b4. The 
> problem is that i am backing up a few kvm images and the biggest one is 
> most of the time active and gets retried 10 times.

If you backup the disk image file of an active VM, the backup will be 
inconsistent. You could as well just not back it up.

You either need to shutdown the VM or use some special measures on the 
VM or fs or block level to at least get "crash like" consistency via 
some sort of snapshot or freezing.

> The new feature of retrying active files needs an option to limit 
> retries to a small number (or maybe including 0 for turning it off). I 
> think an option for a maximum file size for retries would make sense. 
> Retrying a backup of 50 MB sqlite DB is
> much more likely to succeeed then 50GB KVM. But OTOH documentation and 
> usage would be a bit difficult to explain.

Well, we could put the max retries counter into an env var, but as I 
explained above, that would not help you at all, it would just hide the 
severe issue you didn't notice before.

BTW, did't borg 1.2 also warn you all the time for that kvm image that 
the file has changed while being backed up?

> I am running 2.0.0b5 with a few test restores and no problems so far.

Did you also check that "hot" kvm image? Keep in mind that not noticing 
a problem does not necessarily imply that there is no problem.

> I noticed that with borg 2 the cache directories do *not* contain the 
> name borg which is a bit annoying and i would prefer the old behaviour.
> 
> Borg 1.2 
> /home/dietmar/.cache/borg/0a2cb8f22adaa78a1863b028f3d1629393ef0fdd99b3b61c11d871de9eaaee8b/README
> 
> Borg 2.0.0b5 
> /home/dietmar/.cache/fa673b73240ed3b75ba947fd98787fdf8ecd17204323e4a10c2754cdcc5fd547/README

Oh, oops, guess there is a bug. Thanks for reporting that!

**Reply by ThomasWaldmann** [reply]
@dietmargoldbeck to reproduce the bad .cache directory naming, can you please tell us which `BORG_*` and `XDG*` environment variables you have set?

**Reply by ThomasWaldmann** [reply]
@dietmargoldbeck ping.

**Reply by dietmargoldbeck** [reply]
> @dietmargoldbeck to reproduce the bad .cache directory naming, can you please tell us which `BORG_*` and `XDG*` environment variables you have set?

borg is called by a shell script which has the following exports: 
export BORG_REPO
export BORG_REMOTE_PATH=/usr/local/bin/borg-2
export BORG_PASSCOMMAND=foo
BORG_BASE_DIR=${HOME}; export BORG_BASE_DIR
  hope this helps. Sorry for answeringg late.
   Dietmar

**Reply by dietmargoldbeck** [reply]
> If you backup the disk image file of an active VM, the backup will be inconsistent. You could as well just not back it up. You either need to shutdown the VM or use some special measures on the VM or fs or block level to at least get "crash like" consistency via some sort of snapshot or freezing.

If i regularly back up the hot image, all the needed chunks are in borgs repos.
This causes a borg run before upgrades etc of the image to be *fast* and minimizes the time for me to shutdown the
machine before i do major changes. Otherwise i would torture my DSL uplink for a long time.
Maintaining excludes for borg is also error prone. Always backing up all the machine is safer than manually changing excludes depending on what is currently running and what is shutdown.

I am aware of the risk. The VM systems are for software testing and mostly contain git clones and opensource downloads, e.g. borg source code :D.

**Reply by dietmargoldbeck** [reply]
> @dietmargoldbeck to reproduce the bad .cache directory naming, can you please tell us which `BORG_*` and `XDG*` environment variables you have set?

The naming seems to depend on the BORG_BASE_DIR environment variable. I cannot reproduce the bug without exporting BORG_BASE_DIR

**Reply by ThomasWaldmann** [reply]
#7445 is fixed by #7448.
